File: Python/tests/test_reverse_integer.py
import unittest
from reverse_integer import Solution

class ReverseInteger(unittest.TestCase):
    def setUp(self):
        pass
            
    def tearDown(self):
        pass

    def testReverseString(self):
        self.assertEqual(Solution().reverse(4321), 1234)
        self.assertEqual(Solution().reverse(-4321), -1234)
        self.assertEqual(Solution().reverse(1000), 1)
        self.assertEqual(Solution().reverse(1563847412), 0)
        self.assertEqual(Solution().reverse(-1563847412), 0)

File: Ruby/test/test_reverse_integer.rb
require './reverse_integer.rb'
require 'test/unit'

class TestReverseInteger < Test::Unit::TestCase
  def test_reverse_string
    assert_equal(1234, reverse(4321))
    assert_equal(-1234, reverse(-4321))
    assert_equal(1, reverse(1000))
    assert_equal(0, reverse(1_534_236_469))
    assert_equal(0, reverse(1_563_847_412))
    assert_equal(0, reverse(-1_563_847_412))
  end
end
